File: PlaygroundE/Assets/Scripts/Old/Hit.js
#pragma strict


    var damage : int;
    var Speed : float = 1;
    var Target : Vector3;
    var self : GameObject;

    function Update()
    {
        transform.position = Vector3.MoveTowards(transform.position, Target, Speed * Time.deltaTime);
    }

    function OnTriggerEnter(other: Collider)
    {
        if (other.gameObject != self && other.gameObject.tag == "Player")
        {
            var target = other.gameObject.GetComponent.<Stats>();
            target.cur_health -= damage;
            target.setHealthBar();
        }
    }

File: PlaygroundE/Assets/Scripts/Hit.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Hit : MonoBehaviour {

    public float damage;
    public float Speed = 1;
    public GameObject owner;

    void Update()
    {

    }

    void OnTriggerEnter(Collider other)
    {
        if (other.gameObject == owner) {
            return;
        }
        if (other.gameObject.tag == "Player")
        {
            var target = other.gameObject.GetComponent<Stats> ();
            target.Damage(damage);
        }
    }
}
